networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * NetworkitBinaryReader.cpp
+ *
+ *@author Charmaine Ndolo <charmaine.ndolo@b-tu.de>
+ */
+
+#include <networkit/io/NetworkitBinaryReader.hpp>
+
+namespace NetworKit {
+
+size_t NetworkitBinaryReader::decode(const uint8_t* data, uint64_t& value) {
+	int n;
+	if(!data[0]) {
+		n = 8;
+		value = 0;
+	} else {
+		n = tlx::ffs(data[0]) -1;
+		value = data[0] >> (n+1);
+	}
+
+	for(int i = 0; i < n; i++) {
+		value |= data[i+1] << (8 - (n + 1) + i * 8);
+	}
+	return n+1;
+}
+
+int64_t NetworkitBinaryReader::decodeZigzag(uint64_t value) {
+	int64_t orig = (value >> 1) ^ (-(value & 1));
+	return orig;
+}
+
+Graph NetworkitBinaryReader::read(const std::string& path) {
+	Header header = {};
+
+	MemoryMappedFile mmfile(path);
+	auto it = mmfile.cbegin();
+	auto end = mmfile.cend();
+
+	auto readHeader = [&] () {
+		memcpy(&header.magic, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+		memcpy(&header.checksum, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+		memcpy(&header.features, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+		memcpy(&header.nodes, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+		memcpy(&header.chunks, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+		memcpy(&header.offsetBaseData, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+		memcpy(&header.offsetAdjLists, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+		memcpy(&header.offsetWeights, it, sizeof(uint64_t));
+		it += sizeof(uint64_t);
+	};
+
+	auto checkHeader = [&] () {
+		if(memcmp("nkbg000", header.magic, 8)) {
+			throw std::runtime_error("Reader expected another magic value");
+		} else {
+			directed = (header.features & DIR_MASK);
+			weighted = (header.features & WGHT_MASK) >> WGHT_SHIFT;
```

Comment:
With the redefinition of the weight constants, we cannot use the feature field to check whether the graph is weighted anymore. We need to check whether the `offsetWeights` is zero.